### DISCUSSION TITLE
Building Xmonad on Nixos

### DISCUSSION THREAD

**Discussion by jrgiacone** [xmonad/xmonad#393] [Q&A] [answered]
Currently trying to build xmonad:

windowManager.xmonad = {
      enable = true;
      enableContribAndExtras = true;
      extraPackages = haskellPackages: [
        haskellPackages.xmonad_0_17_0
        haskellPackages.xmonad-contrib_0_17_0
        haskellPackages.xmonad-extras_0_17_0
      ];
    };

however, it spits out and error and states xmonad >=0.16.99999 && <0.18. Anyone have any idea how to fix or build from stack, trying to transition from arch to nixos

**Comment by sorki** [comment]
Try a minimal expression like:

```nix
windowManager.xmonad = {
  enable = true;
  enableContribAndExtras = true;
};
```

You don't need to include `xmonad` itself in `extraPackages` and `enableContribAndExtras = true;` will add `xmonad-{contrib,extras}` to `extraPackages` internally.

I'm also wondering why you're using  `xmonad_0_17_0` instead of just `xmonad`.

**Reply by jrgiacone** [reply]
> Try a minimal expression like:
> 
> ```nix
> windowManager.xmonad = {
>   enable = true;
>   enableContribAndExtras = true;
> };
> ```
> 
> You don't need to include `xmonad` itself in `extraPackages` and `enableContribAndExtras = true;` will add `xmonad-{contrib,extras}` to `extraPackages` internally.
> 
> I'm also wondering why you're using `xmonad_0_17_0` instead of just `xmonad`.

Because it is pulling xmonad .15 instead of .17 from the repos even though I thought I enabled unstabel. My config uses stuff from the more recent release

**Reply by sorki** [reply]
Looks like you can also set custom `windowManager.xmonad.haskellPackages` with either `haskellPackages` from unstable or using `haskellPackages.override` to create a version where `xmonad = xmonad_0_17_0` (same for `-contrib` and `-extras`).

**Reply by jrgiacone** [reply]
> Because it is pulling xmonad .15 instead of .17 from the repos even though I thought I enabled unstable. My config uses stuff from the more recent release

I am trying to do that override and getting an error stating attribute 'ghcWithPackages' missing at...

**Comment by jrgiacone** [comment]
@sorki Here are some screenshots, where I import the unstable branch under unstable
![2022-04-22-15:40:18-screenshot](https://user-images.githubusercontent.com/71910145/164813891-2fe0d3ce-4f39-4670-8238-363d48f3df40.png)
![2022-04-22-15:40:35-screenshot](https://user-images.githubusercontent.com/71910145/164813896-dfd3dca1-98b8-4f20-b5ef-2ec80f9d84e0.png)

**Comment by slotThe** [comment, marked as answer]
@jrgiacone did you manage to figure this out?  Btw, we also have a flake and some usage instructions for that in the [contrib repo](https://github.com/xmonad/xmonad-contrib/blob/master/NIX.md)

**Reply by jrgiacone** [reply]
Hi @slotThe , yea I did I ended up just changing the whole system to track unstable which was up to date!
